datanode/gateway/graphql/schema.graphql

```diff
@@ -5706,3 +5724,106 @@ type FundingPeriodConnection {
   "The pagination information"
   pageInfo: PageInfo!
 }
+
+"Referral program information"
+type ReferralProgram {
+  "Unique ID generated from the proposal that created this program."
+  id: ID!
+  "Incremental version of the program. It is incremented each time the referral program is edited."
+  version: Int!
+  "Defined tiers in increasing order. First element will give Tier 1, second element will give Tier 2, etc."
+  benefit_tiers: [BenefitTier!]!
+  "Timestamp as RFC3339Nano, after which when the current epoch ends, the programs status will become STATE_CLOSED and benefits will be disabled."
+  endOfProgramTimestamp: Timestamp!
+  "Number of epochs over which to evaluate a referral set's running volume."
+  windowLength: Int!
+  """
+  Defined staking tiers in increasing order. First element will give Tier 1,
+  second element will give Tier 2, and so on. Determines the level of
+  benefit a party can expect based on their staking.
+  """
+  stakingTiers: [StakingTier!]!
+  "Timestamp as RFC3339Nano when the program ended. If present, the current program has ended and no program is currently running."
+  endedAt: Timestamp
+}
+
+"Data relating to a referral set."
+type ReferralSet {
+  "Unique ID of the created set."
+  id: ID!
+  "Party that created the set."
+  referrer: ID!
+  "Timestamp as RFC3339Nano when the referral set was created."
+  created_at: Timestamp!
+  "Timestamp as RFC3339Nano when the referral set was updated."
+  updated_at: Timestamp!
+  """
+  Referral set statistics for the latest or specific epoch.
+  If provided the results can be filtered for a specific referee
+  """
+  stats(epoch: Int, referee: ID): ReferralSetStats!
+}
+
+"Edge type containing the referral set and cursor information returned by a ReferralSetConnection"
+type ReferralSetEdge {
+  "The referral set"
+  node: ReferralSet!
+  "The cursor for this referral set"
+  cursor: String!
+}
+
+"Connection type for retrieving cursor-based paginated referral set information"
+type ReferralSetConnection {
+  "The referral sets in this connection"
+  edges: [ReferralSetEdge]!
+  "The pagination information"
+  pageInfo: PageInfo!
+}
+
+"Data relating to referees that have joined a referral set"
+type ReferralSetReferee {
+  "Unique ID of the referral set the referee joined."
+  referral_set_id: ID!
+  "Party that joined the set."
+  referee: ID!
```

Comment:
(non blocking): wondering if this would be better named `refereePartyId`, might be clearer